File: 04564.cpp
#include <iostream>
using namespace std;

int main() {
	while (1) {
		int S; cin >> S;
		if (!S) break;

		while (1) {
			cout << S << ' ';
			if (S < 10) break;

			int mul = 1;
			while (S > 0) {
				mul *= S % 10;
				S /= 10;
			}

			S = mul;
		}

		cout << endl;
	}
}

File: 04564.py
while 1:
    S = int(input())
    if not S: break

    while 1:
        print(S, end=' ')
        if S < 10: break

        mul = 1
        while S > 0:
            mul *= S % 10
            S //= 10

        S = mul

    print()
